### PR TITLE
Set transport

### PR DESCRIPTION
**Pull request by costasovo**

### CODE REVIEW

File: src/Trip/Manipulator.test.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { Place } from '../Places';
 import * as PlaceExpectedResults from '../TestData/PlacesExpectedResults';
 import * as TripExpectedResults from '../TestData/TripExpectedResults';
 import * as Manipuator from './Manipulator';
```
Comment (fekecrad) [inline src/Trip/Manipulator.test.ts]: Fix typo probably caused by me :D

File: src/Trip/Manipulator.test.ts [hunk 2]
```diff
@@ -305,4 +305,46 @@ describe('TripManipulator', () => {
...
+		});
+
+		it('should throw an error when invalid itemIndex is passed', () => {
+			const inputTrip: Trip = cloneDeep(TripExpectedResults.tripDetailed);
+
+			return chai.expect(() => Manipuator.setTransport(inputTrip, 0, 999, null))
+				.to.throw(Error, 'Invalid itemIndex');
+		});
+
+		it('should set immutably set correct transport', () => {
```
Comment (fekecrad) [inline src/Trip/Manipulator.test.ts]: Delete one 'set' word

File: src/Trip/Validator.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+export function validateTransportSettings(settings): void {
+	const modes = [
```
Comment (fekecrad) [inline src/Trip/Validator.ts]: can't we use mods/types/avoids from Trip module ?

File: src/Trip/Validator.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	if (!settings.hasOwnProperty('waypoints') && !(settings.waypoints instanceof Array)) {
+		throw new Error('waypoints must be an array');
+	}
+
+	settings.waypoints.forEach((wp) => {
+		if (
+			!wp.hasOwnProperty('lat') ||
+			!wp.hasOwnProperty('lat') ||
```
Comment (NeosinneR) [inline src/Trip/Validator.ts]: Should be `lng`

File: src/Trip/Manipulator.ts [hunk 5]
```diff
@@ -87,6 +87,24 @@ export function swapDaysInTrip(tripToBeUpdated: Trip, firstDayIndex: number, sec
...
+
+	if (!trip.days[dayIndex]) {
+		throw new Error('Invalid dayIndex');
+	}
+
+	if (itemIndex && !trip.days[dayIndex].itinerary[itemIndex]) {
+		throw new Error('Invalid itemIndex');
+	}
+
+	const resultTrip = cloneDeep(trip);
```
Comment (NeosinneR) [inline src/Trip/Manipulator.ts]: Maybe use `let`/`var` since you are changing it on the next line?

**Pull request merged**